### PR TITLE
feat: RainbowKit

### PR DESCRIPTION
**Pull request by colbr**
#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

### CODE REVIEW

File: src/lib/web3/wagmi-config.ts
Comment (colbr) [inline src/lib/web3/wagmi-config.ts:7]: We'll need to set up a project in Wallet Connect Cloud for this. Also will add it to `.env` - have just left it hardcoded for now.

**Comment by colbr** [comment]
@ratik21 @domw30 I'd like some feedback on this one if possible!

It seems like `wagmiConfig` acts like a store for all web3 stuff. Is there a need to keep web3 stuff like `account` in the store, given we can just access it through the `WalletClient` object (which you can get through `wagmiConfig`)?

I couldn't figure out what the web3 events like `web3ChangeAccount` are doing, because when I change account on the main deployment nothing happens 🤷

There's probably a bunch of tidy up that needs to be done here, and some edge cases I might be missing, but it seems like it works. Given signing the login message is the only web3 interaction, I think it's good as is.

File: src/lib/web3/rainbowkit/provider.tsx
Comment (colbr) [inline src/lib/web3/rainbowkit/provider.tsx:18]: [`react-query` provider is required by RainbowKit](https://www.rainbowkit.com/docs/installation#wrap-providers)

**Comment by domw30** [comment]
For reference: 
Screenshot 2024-07-05 at 12 18 26

Screenshot 2024-07-05 at 12 18 42

**Comment by colbr** [comment]
> For reference: Screenshot 2024-07-05 at 12 18 26
> 
> Screenshot 2024-07-05 at 12 18 42

Fixed the "non-serializable" error, but not the reconnect. Let's fix that one next!

**Review by ratik21 — APPROVED** [review]
please also add `REACT_APP_WALLET_CONNECT_PROJECT_ID` to the `.env` before merging

**Review by ratik21 — CHANGES_REQUESTED** [review]
Ok, ran this locally and i am getting this error and a blank screen:
image

File: src/authentication/web3-login/container.test.tsx
Comment (colbr) [reply src/authentication/web3-login/container.test.tsx:27]: `isConnecting` is related to the `login` state, e.g. is the user currently signing the authentication message or waiting for a response from the server. `isWalletConnected` is related to the `web3` state, e.g. there is a wallet connected, but they might not have signed the authentication message yet.

It needs to be handled this way now because RainbowKit is handling our web3 connection state, and we're responding to changes in its internal state, rather than handling the connection ourselves. `isWalletConnected` is required to show the `Sign In` button (which triggers the authentication message signature).

File: src/authentication/create-wallet-account/container.tsx
Comment (colbr) [reply src/authentication/create-wallet-account/container.tsx:56]: My previous comment will give a bit more context on this, but `isConnecting` is related to the `login` state rather than the `web3` state.

File: src/index.tsx
Comment (colbr) [reply src/index.tsx:48]: We could remove them as a separate PR. But, I think it makes sense to remove them here as they have no purpose, and only increase the error surface.

File: src/authentication/create-wallet-account/index.test.tsx
Comment (colbr) [reply src/authentication/create-wallet-account/index.test.tsx:20]: Yeah for sure, I can remove it. I captured it here: https://linear.app/zero-tech/issue/ZOS-165/remove-legacy-walletselect-component

File: src/authentication/web3-login/container.test.tsx
Comment (ratik21) [reply src/authentication/web3-login/container.test.tsx:27]: ok got it

File: src/index.tsx
Comment (ratik21) [reply src/index.tsx:48]: yeah, you can do it here then.

**Review by ratik21 — APPROVED** [review]

**Review by domw30 — APPROVED** [review]
Nice 💯

**Pull request merged**
